### DISCUSSION TITLE
v1.111.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#11455] [Announcements]
# v1.111.0

## Highlights

Welcome to release `v1.111.0` of Immich; this release focuses on UI enhancement, accessibility, and refactoring and adds a new editor feature to the mobile app, which includes photo rotation and cropping. We hope you like it. Some of the key highlights include:

- Photo editor on the mobile app
- Notable fix: Random logout on the mobile app when the connection between the app and server is unstable.

### Rotating and cropping on the mobile app

Thanks to @Yuvi-raj-P, we can rotate and crop a photo on the mobile app. The edited new file will be created, which can be selected for upload to the Immich server or for sharing with others.

This is the first round of implementation of this feature, let's us know your feedback and how do you want this to be improved.

| Edit option is added to the bottom bar | Edit mode (Crop/Rotate) |
| - | - |
|  ![edit-option-bottom-bar](https://github.com/user-attachments/assets/d023cf4a-0d55-4368-a89b-d644707c7206) | ![edit-mode](https://github.com/user-attachments/assets/daab2872-1839-46a4-af93-3953d3a94ed9) |

| Cropping | End result |
| - | - |
|  ![Cropping](https://github.com/user-attachments/assets/0867926b-d6ac-44bf-8bc7-411e70438f19) |  ![end-result](https://github.com/user-attachments/assets/5c74ad36-2e89-4c86-934f-0a509e683e5d) |

## Support Immich

<p align="center">
<img src="https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExbjY2eWc5Y2F0ZW56MmR4aWE0dDhzZXlidXRmYWZyajl1bWZidXZpcyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/87CKDqErVfMqY/giphy.gif" width="450" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich by purchasing a product key at [https://buy.immich.app](https://buy.immich.app).

Cheers! 🍻

### 🗄️ Server
* chore(server): clean mail-templates and add tailwind style by @hitech95 in https://github.com/immich-app/immich/pull/11296
* chore(server): email template minor styling by @alextran1502 in https://github.com/immich-app/immich/pull/11387
* feat(server): significantly improve Australian reverse geocoding accuracy by @lhjt in https://github.com/immich-app/immich/pull/11370
* fix(server): use fqdn for og:image meta tag value by @eleith in https://github.com/immich-app/immich/pull/11082
* feat(server): reverse geocoding endpoint by @jrasm91 in https://github.com/immich-app/immich/pull/11430
* fix(server): correct person birth date across timezones by @michelheusschen in https://github.com/immich-app/immich/pull/11369
### 📱 Mobile
* chore(deps): update dependency flutter to v3.22.3 by @renovate in https://github.com/immich-app/immich/pull/11301
* fix(mobile): search filter translation + fixes by @waclaw66 in https://github.com/immich-app/immich/pull/11141
* feat(mobile): adds crop and rotate to mobile by @Yuvi-raj-P in https://github.com/immich-app/immich/pull/10989
* fix(mobile): client TLS on ios by @alextran1502 in https://github.com/immich-app/immich/pull/11415
* chore(mobile): Add text to bottom gallery bar by @alextran1502 in https://github.com/immich-app/immich/pull/11417
* fix(mobile): mobile logging out randomly by @alextran1502 in https://github.com/immich-app/immich/pull/11431
* chores(mobile): Translations update by @alextran1502 in https://github.com/immich-app/immich/pull/11454
### 🖥️ Web
* fix(web): buy immich translations by @michelheusschen in https://github.com/immich-app/immich/pull/11379
* fix(web): easier alt text translation for other languages by @ben-basten in https://github.com/immich-app/immich/pull/11124
* chore(web): improve responsiveness in Album and Shared Album pages on small devices by @ayykamp in https://github.com/immich-app/immich/pull/11055
* fix(web): responsive design when selecting assets in an album by @martabal in https://github.com/immich-app/immich/pull/11169
* feat(web): search bar keyboard accessibility by @ben-basten in https://github.com/immich-app/immich/pull/11323
* feat(web): Duplicate-Page shortcut changes by @JW-CH in https://github.com/immich-app/immich/pull/11183
* fix(web): slideshow settings title by @michelheusschen in https://github.com/immich-app/immich/pull/11396
* fix(web): translation leftovers by @waclaw66 in https://github.com/immich-app/immich/pull/11412
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/11365
* fix(web): avoid nesting buttons inside links by @michelheusschen in https://github.com/immich-app/immich/pull/11425
* feat(web): more localized number formatting by @michelheusschen in https://github.com/immich-app/immich/pull/11401
* fix(web): timeline group date formatting by @michelheusschen in https://github.com/immich-app/immich/pull/11392
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/11416
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/11429
### 📓 Documentation
* docs: how to change PG PW by @mmomjian in https://github.com/immich-app/immich/pull/11414
* docs: file custom location by @mmomjian in https://github.com/immich-app/immich/pull/11413
### 🔨 Maintenance
* chore(deps): update dependency byte-size to v9 by @renovate in https://github.com/immich-app/immich/pull/11356
* chore(deps): update docker.io/redis:6.2-alpine docker digest to e3b17ba by @renovate in https://github.com/immich-app/immich/pull/11302
* chore(deps): update dependency @types/picomatch to v3 by @renovate in https://github.com/immich-app/immich/pull/11096
* chore(deps): update terraform cloudflare to v4.38.0 by @renovate in https://github.com/immich-app/immich/pull/11423
* chore(deps): update dependency node to v20.16.0 by @renovate in https://github.com/immich-app/immich/pull/11421
* docs: warning to not edit volumes in compose by @mmomjian in https://github.com/immich-app/immich/pull/11432
* chore(deps): update node.js to eb8101c by @renovate in https://github.com/immich-app/immich/pull/11436
* fix(deps): update dependency exiftool-vendored to v28.2.0 by @renovate in https://github.com/immich-app/immich/pull/11439
* chore(deps): update dependency exiftool-vendored to v28 by @renovate in https://github.com/immich-app/immich/pull/11440
* chore(deps): update base-image to v20240730 (major) by @renovate in https://github.com/immich-app/immich/pull/11447
### Other changes
* chore(mobile): post release task by @alextran1502 in https://github.com/immich-app/immich/pull/11382
* chore: bump vitest to 1.6.0 by @etnoy in https://github.com/immich-app/immich/pull/11386
* fix(server): make vitest pick up edited files by @etnoy in https://github.com/immich-app/immich/pull/11385
* chore(server): make vite-tsconfig-paths a dev dependency instead by @etnoy in https://github.com/immich-app/immich/pull/11404

## New Contributors
* @ayykamp made their first contribution in https://github.com/immich-app/immich/pull/11055
* @Yuvi-raj-P made their first contribution in https://github.com/immich-app/immich/pull/10989
* @lhjt made their first contribution in https://github.com/immich-app/immich/pull/11370
* @eleith made their first contribution in https://github.com/immich-app/immich/pull/11082

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.110.0...

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.111.0'>v1.111.0</a>.</em>

**Comment by stephen304** [comment]
This looks so nice! I can't wait for it to land for web too. It would be awesome if the editor could also handle photo edit intents so I can jump directly to cropping for example screenshots using immich instead of google photos.

**Reply by alextran1502** [reply]
The UI for the web is being worked on, and there are a few more steps to be done. I expect it will land in the next release #11074

**Reply by dohlin** [reply]
Sooooo excited to see the mobile photo editing features starting to roll in - I can't wait for this feature to be more fully built out (markup, etc. tools added) as well as for image opening (and editing) intents to be available (i.e. open this picture with Immich instead of GPhotos or another app).  Bought my supporter contribution a couple days back and got a coworker to do the same!  Super exciting.  Keep up the great work!!

**Comment by Yuvi-raj-P** [comment]
Glad to be part of this release, the filters feature in the editor is being worked on currently and hopefully should be out by the next release! Please let me know of any feedback on the editor or feature requests. 😀

**Reply by hoggatt** [reply]
Great work! 

One small piece of feedback: maybe this isn't really plausible given the architecture, but often when I just want to rotate a photo, I don't want to save a copy. I just want to rotate the original. 

Is this something that would be possible to implement?

**Reply by Yuvi-raj-P** [reply]
I could give you the option to save unto the originals or save as a copy if you want, but it does require some work due to directly saving to the server. No guarantees but I will keep it in mind. 🙂

**Reply by zilexa** [reply]
> I could give you the option to save unto the originals or save as a copy if you want, but it does require some work due to directly saving to the server. No guarantees but I will keep it in mind. 🙂

Wannabe Immich user here (just need time to refresh my server first): +1, because 8 out of 10 times the edited file is the only one I want to keep forever. 
But then the edited file needs to have the original timestamp, metadata etc.

**Comment by SL666** [comment]
Hey there, great to see a basic editor start to take shape... but..

When i crop or rotate and 'save to gallery' the changes don't seem to save?
(with updated docker & web app)

edit - Oh, it appears to save a copy to my on-phone gallery folder? but doesn't update the immich one?

**Reply by Yuvi-raj-P** [reply]
It gets automatically loaded to the server when the background backup is invoked, so do you have the background backup turned on? If yes then you should see it get into the server soon as a copy. It does not update the original image in any way.

**Reply by SL666** [reply]
I do have background backup on - but it appears to save to the 'Pictures' folder eg

/storage/emulated/0/Pictures/_edited(3).jpg

I back up camera and a couple of others.

I guess ideally it would save to the server as the original file name appending perhaps an edit time? even if that was via a specially designated folder that was then pushed using the existing backup process?

**Reply by samolego** [reply]
Oh, now I see this; I opened a bug #11472 for that

**Reply by obermotz** [reply]
Also the edited image does not preserve the metadata/exif info of the original. In it's current state the editor is of no use (to me), but hopefully will develop into something useful.

Edit: did not see replies covering the same issue, shame on me :)

**Comment by suderman** [comment]
Works well, except the location data (and perhaps all metadata?) in the cropped copy is lost.

**Reply by JordyEGNL** [reply]
This is the same feedback I would've given :)

**Reply by Yuvi-raj-P** [reply]
Glad to hear that, working on the metadata should be out by next week or so!

**Comment by agentgodzilla** [comment]
Are developers open to the idea of having the image selection/size in cropping tool to start at full size of the image rather than the current default?  

In my mind cropping is removing parts of an image, so me as an "editor" having to actually start by expanding the selection doesn't make sense.

Also I struggle with expanding the selection to full width with my thumb due to phone case making me have to stop holding phone with one hand to do a very basic crop.

How it is currently
![current](https://github.com/user-attachments/assets/b00709ef-a594-4d32-9234-c40686bcef95)

How I would like it
![hopeful](https://github.com/user-attachments/assets/0ace045a-6927-458c-bef2-e6adb8643adb)

**Reply by jrasm91** [reply]
If it starts out as full it might be hard to see the interface to begin. Maybe we could add padding to the image though, so it doesn't take up the full width or height when it is being cropped.

**Reply by stephen304** [reply]
> If it starts out as full it might be hard to see the interface to begin. Maybe we could add padding to the image though, so it doesn't take up the full width or height when it is being cropped.

Adding padding is how google photos and my gallery app does cropping, with the default crop being the whole image.

**Reply by Yuvi-raj-P** [reply]
I have already tried doing the padding, but the problem we had was it is making the image harder to view (because images gets smaller and landscapes are more hard if they're small). Secondly if the crop rectangle is extended completely to the screen on default it makes cropping very hard due to the crop corners being close to the screen edges making it harder to move the rectangle. I do have no problem working on this change but if we have more people requesting this change I would consider it.

**Reply by stephen304** [reply]
Good point, one thing google does that makes it better for cropping small is it zooms in to the cropped area, so the padding goes away when it zooms so you don't end up with a hard to control small cropping rectangle. That doesn't sound trivial to implement though.

**Reply by Yuvi-raj-P** [reply]
Yeah, that requires a completely newly designed custom crop controller which is going to take a lot of effort and time. I am planning on getting the basics first then going on with refining and adding more features to the existing ones. So yeah completely different story for later 😁

**Reply by agentgodzilla** [reply]
Maybe for time being add another option in the aspect ratio options in the bottom when cropping labeled "Full" or something like that which quickly expands the selection to the full image?  Can worry less about visibility concerns as users have to select it and it doesn't need some large refactor.

If worried about breaking "design principles" where buttons in a row don't all do the same thing as it's the same AR as the free option, could also do another row of buttons that have basic 100%, 75%, 50%, 25% size options that apply to all ARs

**Reply by jrasm91** [reply]
Maybe don't do any padding on landscape then. But I think minimal padding would go a long way otherwise

**Comment by coxde** [comment]
Maybe the buttons would be more ergonomic (at least for right-handed) if the order is like this (Delete, Archive, Edit and Share).

**Reply by samolego** [reply]
I'd argue it's good as it is, as the-app-which-must-not-be-named uses same layout

**Comment by 0rn0lf** [comment]
Just for calrification, when we edit an image it will be saved as a new image without the old metadata?
Will the old image be deleted by the server?

**Reply by JordyEGNL** [reply]
The old image will not be removed or changed

**Comment by JackBailey** [comment]
It'd be nice as an option so that when you drag the cropping box to crop it, it'll zoom in so the new crop box fills up most of the screen. Like Samsung and Google's Photo apps.

**Reply by ZtereoHYPE** [reply]
yeah, currently it seems to take the minimum edge size but that leads to a bug where if you switch between ratios the area keeps shrinking

**Reply by Yuvi-raj-P** [reply]
That feature does require a lot of work and a completely new crop controller but the banking is Fixed #11462

**Comment by AngelaDMerkel** [comment]
I feel like it would be nice if only a new preview thumb was generated at the cropped size rather than there being a duplicate image produced, but I am not sure this is the behaviour others are looking for

**Comment by crypto-smoke** [comment]
A bit of a small request, but it would be great if the changelog items could be organized so that we can quickly see new features and bugfixes by having them grouped together. For example, instead of the random sorting in each category:

🗄️ Server

- chore(server): clean mail-templates and add tailwind style by @hitech95 in https://github.com/immich-app/immich/pull/11296
- chore(server): email template minor styling by @alextran1502 in https://github.com/immich-app/immich/pull/11387
- feat(server): significantly improve Australian reverse geocoding accuracy by @lhjt in https://github.com/immich-app/immich/pull/11370
- fix(server): use fqdn for og:image meta tag value by @eleith in https://github.com/immich-app/immich/pull/11082
- feat(server): reverse geocoding endpoint by @jrasm91 in https://github.com/immich-app/immich/pull/11430
- fix(server): correct person birth date across timezones by @michelheusschen in https://github.com/immich-app/immich/pull/11369

We could have them sorted by feature, fix, chore:

🗄️ Server

- feat(server): significantly improve Australian reverse geocoding accuracy by @lhjt in https://github.com/immich-app/immich/pull/11370
- feat(server): reverse geocoding endpoint by @jrasm91 in https://github.com/immich-app/immich/pull/11430
- fix(server): use fqdn for og:image meta tag value by @eleith in https://github.com/immich-app/immich/pull/11082
- fix(server): correct person birth date across timezones by @michelheusschen in https://github.com/immich-app/immich/pull/11369
- chore(server): clean mail-templates and add tailwind style by @hitech95 in https://github.com/immich-app/immich/pull/11296
- chore(server): email template minor styling by @alextran1502 in https://github.com/immich-app/immich/pull/11387

**Reply by jrasm91** [reply]
I've been meaning to do this for awhile, but #11726 should be better. Basically opt-in to having commits be shown at the top of the release notes as features or bug fixes and everything else gets thrown in at the bottom.
